matlab/src/cpp/arrow/matlab/proxy/factory.cc

```diff
@@ -28,6 +28,10 @@ std::shared_ptr<Proxy> Factory::make_proxy(const ClassName& class_name, const Fu
     // Register MATLAB Proxy classes with corresponding C++ Proxy classes.
     REGISTER_PROXY(arrow.array.proxy.Float32Array, arrow::matlab::array::proxy::NumericArray<float>);
     REGISTER_PROXY(arrow.array.proxy.Float64Array, arrow::matlab::array::proxy::NumericArray<double>);
+    REGISTER_PROXY(arrow.array.proxy.Int8Array, arrow::matlab::array::proxy::NumericArray<int8_t>);
+    REGISTER_PROXY(arrow.array.proxy.Int16Array, arrow::matlab::array::proxy::NumericArray<int16_t>);
+    REGISTER_PROXY(arrow.array.proxy.Int32Array, arrow::matlab::array::proxy::NumericArray<int32_t>);
+    REGISTER_PROXY(arrow.array.proxy.Int64Array, arrow::matlab::array::proxy::NumericArray<int64_t>);
```

Comment:
We may want to use the same style with `UInt*`:

https://github.com/apache/arrow/blob/4129fe1a3e6fca1ab46d9f59a914110f58ec94c6/matlab/src/cpp/arrow/matlab/proxy/factory.cc#L28-L32

Or we may want to put all `REGISTER_PROXY()` to here:

```cpp
    // Register MATLAB Proxy classes with corresponding C++ Proxy classes.
    REGISTER_PROXY(arrow.array.proxy.Float32Array, arrow::matlab::array::proxy::NumericArray<float>);
    REGISTER_PROXY(arrow.array.proxy.Float64Array, arrow::matlab::array::proxy::NumericArray<double>);
    REGISTER_PROXY(arrow.array.proxy.Int8Array, arrow::matlab::array::proxy::NumericArray<int8_t>);
    REGISTER_PROXY(arrow.array.proxy.Int16Array, arrow::matlab::array::proxy::NumericArray<int16_t>);
    REGISTER_PROXY(arrow.array.proxy.Int32Array, arrow::matlab::array::proxy::NumericArray<int32_t>);
    REGISTER_PROXY(arrow.array.proxy.Int64Array, arrow::matlab::array::proxy::NumericArray<int64_t>);
    REGISTER_PROXY(arrow.array.proxy.UInt8Array, arrow::matlab::array::proxy::NumericArray<uint8_t>);
    REGISTER_PROXY(arrow.array.proxy.UInt16Array, arrow::matlab::array::proxy::NumericArray<uint16_t>);
    REGISTER_PROXY(arrow.array.proxy.UInt32Array, arrow::matlab::array::proxy::NumericArray<uint32_t>);
    REGISTER_PROXY(arrow.array.proxy.UInt64Array, arrow::matlab::array::proxy::NumericArray<uint64_t>);
```